setup.py

```diff
@@ -127,9 +127,11 @@ def read(*rnames):
       headers=headers,
       ext_modules=ext_modules,
       python_requires='>=3.9',
+      setup_requires=[
+          'setuptools',
```

Comment:
`setup_requires=['setuptools']` is always pointless: the thing that evaluates `setup_requires` is always setuptools itself, so it can't possibly not be installed.  Once setuptools is no longer needed at run-time, you should just drop it from `install_requires` without adding it to `setup_requires`.